### PR TITLE
[MIG][11.0] hr_timesheet: migration scripts

### PR DESCRIPTION
**Pull request by cubells**
Depends on:
- [x] sale_timesheet

cc @Tecnativa

### CODE REVIEW

**Comment by StefanRijnhart** [comment]
Looks like you need to cover sale_timesheet as well here to make travis green, or at least that field 'is_service' which you need to precreate and fill with an SQL query.

**Comment by cubells** [comment]
hi @StefanRijnhart I'm working on it.

File: addons/hr_timesheet/migrations/11.0.1.0/openupgrade_analysis_work.txt [hunk 1]
```diff
@@ -0,0 +1,74 @@
+---Fields in module 'hr_timesheet'---
+hr_timesheet / account.analytic.line    / employee_id (many2one)        : NEW relation: hr.employee
```
Comment (pedrobaeza) [inline addons/hr_timesheet/migrations/11.0.1.0/openupgrade_analysis_work.txt:2]: Fill this with the SQL equivalent to:
`self.env['hr.employee'].search([('user_id', '=', result['user_id'])], limit=1).id`

File: addons/hr_timesheet/migrations/11.0.1.0/openupgrade_analysis_work.txt [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+# Nothing to do
+
+hr_timesheet / project.project          / subtask_project_id (many2one) : module is now 'project' ('hr_timesheet')
+# Nothing to do: it must be done in project migration
```
Comment (pedrobaeza) [inline addons/hr_timesheet/migrations/11.0.1.0/openupgrade_analysis_work.txt]: You can group all these lines together with the same comment for better reading

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
`sale_timesheet` gets auto-installed when installing this one, so let's see if we can merge sale_timesheet to get this one green.

File: addons/hr_timesheet/migrations/11.0.1.0/openupgrade_analysis_work.txt (hunk 1 above)
Comment (MiquelRForgeFlow) [reply addons/hr_timesheet/migrations/11.0.1.0/openupgrade_analysis_work.txt:2]: Yes, this must be filled with this.

File: addons/hr_timesheet/migrations/11.0.1.0/pre-migration.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+_xmlid_renames = [
+    ('website_project_timesheet.portal_task_timesheet_rule',
+     'hr_timesheet.timesheet_rule_portal'),
+]
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.rename_xmlids(env.cr, _xmlid_renames)
```
Comment (MiquelRForgeFlow) [inline addons/hr_timesheet/migrations/11.0.1.0/pre-migration.py:14]:
```python
if openupgrade.table_exists(cr, 'website'):
```

Comment (pedrobaeza) [reply addons/hr_timesheet/migrations/11.0.1.0/pre-migration.py:14]: No need of that, as the rename won't have any effect if non existant

Comment (MiquelRForgeFlow) [reply addons/hr_timesheet/migrations/11.0.1.0/pre-migration.py:14]: :+1:

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**